plugin/cache/handler.go

```diff
@@ -89,6 +89,15 @@ func (c *Cache) exists(state request.Request) *item {
 	return nil
 }
 
+// CacheResetMetrics - ensure all metrics for cache re-start from 0
+func CacheResetMetrics() {
```

Comment:
when is this needed?

Reply:
all the metrics collectors are declared as global vars.
All our UT or Tests are running on the same process. 
If you wish to test the value of these metrics inside one of the tests, you need to Reset the metrics before running that specific test, else you got whatever other test may have used.

I am introducing such a test with `TestMetricsSeveralBlocs`.
That was also the reason (IMO) of the existing test `TestMetricsCache` that was flaky.

That is also related to this issue #2206

Reply:
ok, but then it does not have to be defined in the main code; just a test helper function.

Reply:
> ok, but then it does not have to be defined in the main code; just a test helper function.

There is an issue with the visibility of the metrics collectors - they are private to the cache package. 

The test function are the test CI functions (not UT that are in the same package).
Unless I miss something, I have either to keep this ResetMetrics() function here, wither to change the collectors to make then public.

Am I correct ?

Reply:
sorry, i don't understand. If it needed in the tests it needs to put there, not here.

Reply:
I need a public function that can reset the metrics collectors that are private to the `cache` package.